lib/openstax/aws/image.rb

```diff
@@ -0,0 +1,26 @@
+module OpenStax::Aws
+  class Image
+
+    attr_reader :image
+
+    def initialize(id:, region:)
+      @image = Aws::EC2::Image.new(id, region: region)
+    end
+
+    def get_tag(key:)
+      tag = image.tags.find{|tag| tag.key == key}
+      raise "No tag with key #{key} on AMI #{image.image_id}" if tag.nil?
+      tag.value
+    end
+
+    def sha
+      get_tag(key: "sha")
+    end
+
+    def self.find_by_sha(sha:, region:)  
+      Aws::EC2::Client.new(region: region).describe_images({
+        filters: [{name: "tag:sha", values: [sha]}]
+      }).images
```

Comment:
I think it would make more since for this `OpenStax::Aws::Image` class to return `OpenStax::Aws::Image` objects here.  Can you change `.images` to `.images.map{|aws_image| OpenStax::Aws::Image.new(aws_image: aws_image)}` which means you'd also need to change the initializer to accept an `aws_image` argument optionally.

Reply:
since this is already a class method, you can probably just say `new` instead of `OpenStax::Aws::Image.new`